Bottom: a0fe51fce1a37b24a00ee0fce84c819b867120f8
Top:    47f4ba42dbf4edd32ee03bbf1b8b4527162c94f5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-10 21:10:04 +0000

Refresh of toy-rendering-attempt

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e9fa5fc..2dff37d 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2135,12 +2135,14 @@ hidgl_draw_step_model_instance (struct assembly_model_instance *instance)
                 STEP_TO_COORD_Y (PCB, instance->oy),
                 STEP_TO_COORD_Z (PCB, instance->oz));
 
-  // XXX: Not sure why we need to flip the instance "y" values, this is strange
-  ox = -instance->ay * instance->rz - instance->az * -instance->ry;
+  /* Undo -Y coord scaling */
+  glScalef (1.0f, -1.0f, 1.0f);
+
+  ox = instance->ay * instance->rz - instance->az * -instance->ry;
   oy = instance->az * instance->rx - instance->ax * instance->rz;
-  oz = instance->ax * -instance->ry - -instance->ay * instance->rx;
+  oz = instance->ax * instance->ry - instance->ay * instance->rx;
   m[0][0] = instance->rx;  m[1][0] = ox;    m[2][0] = instance->ax;    m[3][0] = 0.0f;
-  m[0][1] = -instance->ry;  m[1][1] = oy;    m[2][1] = -instance->ay;    m[3][1] = 0.0f;
+  m[0][1] = instance->ry;  m[1][1] = oy;    m[2][1] = instance->ay;    m[3][1] = 0.0f;
   m[0][2] = instance->rz;  m[1][2] = oz;    m[2][2] = instance->az;    m[3][2] = 0.0f;
   m[0][3] = 0.0f;          m[1][3] = 0.0f;  m[2][3] = 0.0f;            m[3][3] = 1.0f;
   glMultMatrixf(&m[0][0]);
@@ -2166,6 +2168,9 @@ hidgl_draw_step_model_instance (struct assembly_model_instance *instance)
   m[3][0] = 0.0f;            m[3][1] = 0.0f;  m[3][2] = 0.0f;            m[3][3] = 1.0f;
   glMultMatrixf(&m[0][0]);
 
+  /* Undo -Y coord scaling */
+  glScalef (1.0f, -1.0f, 1.0f);
+
   glTranslatef (-STEP_TO_COORD_X (PCB, step_model->ox),
                 -STEP_TO_COORD_Y (PCB, step_model->oy),
                 -STEP_TO_COORD_Z (PCB, step_model->oz));
